src/driver/sqlite-abstract/AbstractSqliteQueryRunner.ts

```
@@ -456,26 +456,32 @@ export abstract class AbstractSqliteQueryRunner extends BaseQueryRunner implemen
         const table = tableOrName instanceof Table ? tableOrName : await this.getCachedTable(tableOrName);
 
         // clone original table and remove column and its constraints from cloned table
         const changedTable = table.clone();
-        columns.forEach(column => {
-            changedTable.removeColumn(column);
-            changedTable.findColumnUniques(column).forEach(unique => changedTable.removeUniqueConstraint(unique));
-            changedTable.findColumnIndices(column).forEach(index => changedTable.removeIndex(index));
-            changedTable.findColumnForeignKeys(column).forEach(fk => changedTable.removeForeignKey(fk));
+        columns.forEach((column: TableColumn|string) => {
+            const columnInstance = column instanceof TableColumn ? column : table.findColumnByName(column);
+            if (!columnInstance) return;
+
+            changedTable.removeColumn(columnInstance);
+            changedTable.findColumnUniques(columnInstance).forEach(unique => changedTable.removeUniqueConstraint(unique));
+            changedTable.findColumnIndices(columnInstance).forEach(index => changedTable.removeIndex(index));
+            changedTable.findColumnForeignKeys(columnInstance).forEach(fk => changedTable.removeForeignKey(fk));
         });
 
         await this.recreateTable(changedTable, table);
 
         // remove column and its constraints from original table.
-        columns.forEach(column => {
-            table.removeColumn(column);
-            table.findColumnUniques(column).forEach(unique => table.removeUniqueConstraint(unique));
-            table.findColumnIndices(column).forEach(index => table.removeIndex(index));
-            table.findColumnForeignKeys(column).forEach(fk => table.removeForeignKey(fk));
+        columns.forEach((column: TableColumn|string) => {
+            const columnInstance = column instanceof TableColumn ? column : table.findColumnByName(column);
+            if (!columnInstance) return;
```

We must throw an error here instead of silently break the operation:
```ts
if (!columnInstance)
   throw new Error(`Column "${column}" was not found in table "${table.name}"`);
```